writer.go

```diff
@@ -912,14 +912,14 @@ func (b *batchQueue) Close() {
 	b.closed = true
 }
 
-func newBatchQueue(initialSize int) batchQueue {
+func newBatchQueue(initialSize int) *batchQueue {
 	bq := batchQueue{
 		queue: make([]*writeBatch, 0, initialSize),
 	}
 
 	bq.cond.L = &bq.mutex
 
-	return bq
+	return &bq
```

Comment:
The existing code is concerning indeed, I wonder if it works just because the function gets inlined... TBD.

I can see two options:
- change the type of `queue` to `*batchQueue` so we don't make a copy
- have an `init` method on `batchQueue` instead of using a constructor

Reply:
FWIW per the docs. Mutexes mustn't  be copied after their first use.

Reply:
Correct, that's what `go vet` was complaining about:

```
./writer.go:922:9: return copies lock value: github.com/segmentio/kafka-go.batchQueue contains sync.Mutex
```

To be clear, the problem is that returning a `batchQueue` value copies the mutex within. (Perhaps that mutex field should be a pointer ...) The `queue` field doesn't seem to be a problem.

Reply:
OK! I took a different tack. Rather than having this function return a pointer, I instead changed the types of the mutex and cond fields in batchQueue to pointers. That seems like a much more focused fix for the problem. @achille-roussel and @rhansen2 , what do you think?